src/main/java/net/openhft/chronicle/bytes/BytesStore.java

```diff
@@ -397,12 +397,19 @@ default char charAt(int index)
     }
 
     /**
-     * Not supported.
+     * This method is inherited from CharSequence.
+     *
+     * @return  {@link BytesStore} of (end - start) length
+     * containing chars from #charAt(start) to #charAt(end - 1), inclusive.
+     * Memory storage type (heap or native) is preserved.
      */
     @NotNull
     @Override
     default CharSequence subSequence(int start, int end) {
-        throw new UnsupportedOperationException("todo");
+        if (start < 0 || end > length() || end < start)
+            throw new IllegalArgumentException("start " + start + ", end " + end + ", length " + length());
+
+        return subBytes(readPosition() + start, end - start);
```

Comment:
use (long)end to perform operations using `long` rather than `int`

Reply:
We know in advance that `end >= start`. Is there another reason to cast except for overflow protection?